### PR TITLE
feat: TTL support for ScyllaDB config and FeatureViews

### PR DESCRIPTION
**Pull request by zabarn**
# What this PR does / why we need it:

Adds TTL support at the feature_store.yaml level, as well as at the FeatureView level when using ScyllaDB as the online store.

# Which issue(s) this PR fixes:

https://jira.expedia.biz/browse/EAPC-15031

# Misc

### CODE REVIEW

File: sdk/python/feast/infra/online_stores/contrib/cassandra_online_store/cassandra_online_store.py [hunk 1]
```diff
@@ -566,8 +570,18 @@ def _create_table(self, config: RepoConfig, project: str, table: FeatureView):
         session: Session = self._get_session(config)
         keyspace: str = self._keyspace
         fqtable = CassandraOnlineStore._fq_table_name(keyspace, project, table)
-        create_cql = self._get_cql_statement(config, "create", fqtable)
-        logger.info(f"Creating table {fqtable}.")
+
+        ttl = (
+            table.online_store_ttl
+            if table.online_store_ttl is not None
+            else config.online_store.ttl
```
Comment (vanitabhagwat) [inline sdk/python/feast/infra/online_stores/contrib/cassandra_online_store/cassandra_online_store.py:577]: Are we using the ttl set in feature_store.yaml here if there is no ttl set at featureview/table level?

Comment (zabarn) [reply sdk/python/feast/infra/online_stores/contrib/cassandra_online_store/cassandra_online_store.py:577]: Yep, that's what this is doing

File: sdk/python/feast/feature_view.py [hunk 2]
```diff
@@ -494,3 +494,23 @@ def most_recent_end_time(self) -> Optional[datetime]:
...
+        Retrieves the online store TTL from the FeatureView's tags.
+
+        Returns:
+            An integer representing the TTL in seconds, or None if not set.
+        """
+        ttl_str = self.tags.get("online_store_ttl")
+        if ttl_str:
+            try:
+                ttl_seconds = int(ttl_str)
+                return ttl_seconds
```
Comment (EXPEbdodla) [inline sdk/python/feast/feature_view.py:510]:
```suggestion
                return int(ttl_str)
```

File: sdk/python/feast/infra/online_stores/contrib/cassandra_online_store/cassandra_online_store.py [hunk 3]
```diff
@@ -88,7 +88,8 @@
         event_ts        TIMESTAMP,
         created_ts      TIMESTAMP,
         PRIMARY KEY ((entity_key), feature_name)
-    ) WITH CLUSTERING ORDER BY (feature_name ASC);
+    ) WITH CLUSTERING ORDER BY (feature_name ASC)
+    {table_options};
```
Comment (EXPEbdodla) [inline sdk/python/feast/infra/online_stores/contrib/cassandra_online_store/cassandra_online_store.py:92]: We can use this way when we add support for multiple table options. At this point, can we just change it to `AND default_time_to_live={ttl}`. Pass 0 when ttl is None.

File: sdk/python/feast/feature_view.py [hunk 4]
```diff
@@ -494,3 +494,23 @@ def most_recent_end_time(self) -> Optional[datetime]:
         if len(self.materialization_intervals) == 0:
             return None
         return max([interval[1] for interval in self.materialization_intervals])
+
+    @property
+    def online_store_ttl(self) -> Optional[int]:
```
Comment (EXPEbdodla) [inline sdk/python/feast/feature_view.py:499]: Please add tests

File: sdk/python/feast/infra/online_stores/contrib/cassandra_online_store/cassandra_online_store.py [hunk 5]
```diff
@@ -88,7 +88,8 @@
         event_ts        TIMESTAMP,
         created_ts      TIMESTAMP,
         PRIMARY KEY ((entity_key), feature_name)
-    ) WITH CLUSTERING ORDER BY (feature_name ASC);
+    ) WITH CLUSTERING ORDER BY (feature_name ASC)
```
Comment (EXPEbdodla) [inline sdk/python/feast/infra/online_stores/contrib/cassandra_online_store/cassandra_online_store.py:91]: Add tests if there are any.

**Review by EXPEbdodla — COMMENTED** [review]
Can we also add the ttl at feature view level for Redis online store as well?

File: sdk/python/tests/integration/feature_repos/repo_configuration.py [hunk 6]
```diff
@@ -71,6 +71,9 @@
 from tests.integration.feature_repos.universal.online_store.bigtable import (
     BigtableOnlineStoreCreator,
 )
+from tests.integration.feature_repos.universal.online_store.cassandra import (
```
Comment (EXPEbdodla) [inline sdk/python/tests/integration/feature_repos/repo_configuration.py:74]: Revert the changes for now. We may need to follow the documentation in docs/how-to-guides/customizing-feast/adding-support-for-a-new-online-store.md to support integration tests.

File: sdk/python/feast/feature_view.py [hunk 7]
```diff
@@ -494,3 +494,22 @@ def most_recent_end_time(self) -> Optional[datetime]:
...
+        """
+        Retrieves the online store TTL from the FeatureView's tags.
+
+        Returns:
+            An integer representing the TTL in seconds, or None if not set.
+        """
+        ttl_str = self.tags.get("online_store_ttl")
```
Comment (EXPEbdodla) [inline sdk/python/feast/feature_view.py:506]: Change the tag name to `online_store_key_ttl_seconds`. For easy to remember and align with feature_store.yaml configuration.

File: sdk/python/feast/infra/online_stores/contrib/cassandra_online_store/cassandra_online_store.py [hunk 8]
```diff
@@ -159,6 +159,9 @@ class CassandraOnlineStoreConfig(FeastConfigBaseModel):
     Table deletion is not currently supported in this mode.
     """
 
+    key_ttl_seconds: Optional[StrictInt] = None
+    """Default TTL (in seconds) to apply to all tables if not specified in FeatureView."""
```
Comment (EXPEbdodla) [inline sdk/python/feast/infra/online_stores/contrib/cassandra_online_store/cassandra_online_store.py:163]:
```suggestion
    """Default TTL (in seconds) to apply to all tables if not specified in FeatureView. Value 0 means No TTL."""
```

**Review by EXPEbdodla — APPROVED** [review]
